internal/parser/planparserv2/pattern_match.go

```diff
@@ -0,0 +1,89 @@
+package planparserv2
+
+import (
+	"fmt"
+
+	"github.com/milvus-io/milvus/internal/proto/planpb"
+)
+
+// Note: just like PostgreSQL, we just plan to support two wildcard in pattern match: `_` and `%`.
+
+var wildcards = map[byte]struct{}{
+	'_': {},
+	'%': {},
+}
+
+var escapeCharacter byte = '\\'
+
+// hasWildcards returns true if pattern contains any wildcard.
+func hasWildcards(pattern string) bool {
+	l := len(pattern)
+	i := l - 1
+	for ; i >= 0; i-- {
+		_, ok := wildcards[pattern[i]]
+		if ok {
+			if i > 0 && pattern[i-1] == escapeCharacter {
+				i--
+				continue
+			}
+			return true
+		}
+	}
+	return false
+}
+
+// findLocOfLastWildcard find the location of last wildcard.
+func findLocOfLastWildcard(pattern string) int {
+	loc := len(pattern) - 1
+	for ; loc >= 0; loc-- {
+		_, ok := wildcards[pattern[loc]]
+		if !ok {
+			break
+		}
+		if ok {
+			if loc > 0 && pattern[loc-1] == escapeCharacter {
+				break
+			}
+		}
+	}
+	return loc
+}
+
+// patternToPrefixMatch translates pattern to a prefix match,
+// error is reported if pattern is not a prefix match.
+func patternToPrefixMatch(pattern string) (string, error) {
+	loc := findLocOfLastWildcard(pattern)
+	if loc < 0 {
+		// always match.
+		return "", nil
+	}
+	if loc >= len(pattern)-1 {
+		return "", fmt.Errorf("pattern is not of prefix match: %s", pattern)
+	}
+	if hasWildcards(pattern[:loc+1]) {
+		return "", fmt.Errorf("pattern is not of prefix match: %s", pattern)
+	}
+	return pattern[:loc+1], nil
+}
+
+// patternToEqualMatch translates pattern to a equal match,
+// error is reported if pattern is not a equal match.
+func patternToEqualMatch(pattern string) (string, error) {
+	if !hasWildcards(pattern) {
+		return pattern, nil
+	}
+	return "", fmt.Errorf("pattern is not of equal match: %s", pattern)
+}
+
+// translatePatternMatch translates pattern to related op type and operand.
+func translatePatternMatch(pattern string) (op planpb.OpType, operand string, err error) {
+	prefix, err := patternToPrefixMatch(pattern)
+	if err == nil {
+		return planpb.OpType_PrefixMatch, prefix, nil
+	}
+	equalStr, err := patternToEqualMatch(pattern)
+	if err == nil {
+		return planpb.OpType_Equal, equalStr, nil
+	}
+	return planpb.OpType_Invalid, "", fmt.Errorf("unsupported pattern: %s", pattern)
```

Comment:
add more error info, for example: "only ab% prefix like is supported"

Reply:
added